src/Hasher.php

```diff
@@ -0,0 +1,53 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Utils;
+
+use EoneoPay\Utils\Exceptions\HashingFailedException;
+use EoneoPay\Utils\Interfaces\HasherInterface;
+
+class Hasher implements HasherInterface
+{
+    /**
+     * @param string $string
+     * @param null|integer $cost
+     *
+     * @return string
+     *
+     * @throws \EoneoPay\Utils\Exceptions\HashingFailedException
+     */
+    public function hash(string $string, ?int $cost = null): string
+    {
+        $options = [];
+
+        // Only set cost if specified, otherwise algorithm default is used
+        if ($cost) {
```

Comment:
If cost isn't set we should set one, if we don't it will be set to 10 anyway, then you can simplify the method call and remove all the lines above:
$hash = \password_hash($string, self::ALGO, ['cost' => $cost ?? self::DEFAULT_COST]);